tests/junit.py

```diff
@@ -6,12 +6,21 @@
 from junit_xml import TestCase, TestSuite
 
 def parse_testargs(file):
+    allargs = []
     if os.path.splitext(file)[1] in ['.c', '.cpp']:
-        return sum([[line.split()[1:]] for line in open(file).readlines()
-                    if line.startswith('//TESTARGS')], [])
+        for line in open(file).readlines():
+            if line.startswith('//TESTARGS'):
+                args = [line.split()[1:]]
+                args.append([line.split()[0].strip('//TESTARGS(name=').strip(')')])
+                allargs.append(args)
```

Comment:
What about 
```
return sum([[line.split()[1:] + line.split()[0].strip('C_TESTARGS(name=').strip(')')
            for line in open(file).readlines()
            if line.startswith('//TESTARGS')], [])
```
(I think I got the syntax right)

Reply:
Hmm! So, I have stored names in a separate sublist for each test to be able to retrieve them [here](https://github.com/CEED/libCEED/blob/4cc1b4280947a15993ad46cc05fb6c1cbb44106c/tests/junit.py#L72) and print them [here](https://github.com/CEED/libCEED/blob/4cc1b4280947a15993ad46cc05fb6c1cbb44106c/tests/junit.py#L92) if there is a name. Are you thinking of a better approach?

Reply:
```
return sum([[[line.split()[1:]] + [line.split()[0].strip('C_TESTARGS(name=').strip(')')]
            for line in open(file).readlines()
            if line.startswith('//TESTARGS')], [])
```
No, I was just trying to make the syntax more compact. Maybe that syntax?

Reply:
Thank you! Unfortunately, none of them work.

Reply:
```
def parse_testargs(file):
    if os.path.splitext(file)[1] in ['.c', '.cpp']:
        return sum([[[line.split()[1:], [line.split()[0].strip('//TESTARGS(name=').strip(')')]]]
                    for line in open(file).readlines()
                    if line.startswith('//TESTARGS')], [])
        return allargs
    elif os.path.splitext(file)[1] == '.usr':
        return sum([[[line.split()[1:], [line.split()[0].strip('C_TESTARGS(name=').strip(')')]]]
                    for line in open(file).readlines()
                    if line.startswith('C_TESTARGS')], [])
    raise RuntimeError('Unrecognized extension for file: {}'.format(file))
```
This one?

Reply:
Beautiful! It worked, thanks!